src/reducers/entities/groups.ts

```diff
@@ -28,13 +28,14 @@ function syncables(state: Dictionary<GroupSyncables> = {}, action: GenericAction
             },
         };
     }
+    case GroupTypes.PATCHED_GROUP_TEAM:
     case GroupTypes.LINKED_GROUP_TEAM: {
         let nextGroupTeams: GroupTeam[] = [];
-
-        if (!state[action.data.group_id] || !state[action.data.group_id].teams) {
+        if (!state[action.data.group_id] || !state[action.data.group_id].teams || state[action.data.group_id].teams.length === 0) {
             nextGroupTeams = [action.data];
         } else {
-            nextGroupTeams = {...state}[action.data.group_id].teams;
+            nextGroupTeams = {...state}[action.data.group_id].teams.slice();
```

Comment:
I'm confused, why do we do the `.slice()` ?

Reply:
Not a blocker so I approved but I'm not quite clear on why the `.slice()`. Especially cause we don't pass anything into it.

Reply:
@hahmadia, slice() creates a new object. I was hitting this weird issue where I could not update the `nextGroupTeams` if I recall correctly, probably because it is a subset of `state`(?). The way out was to create a new version of it to be updated for the next state.

Reply:
If you need a copy of the array, why not do something like 
`[...{...state}[action.data.group_id].teams]`. Does this not work?

Reply:
could be, but slice() was the existing pattern - see `UNLINKED_GROUP_TEAM`, `UNLINKED_GROUP_CHANNEL` , not sure of the pros/cons of using one vs. the other.

Reply:
I'm not sure of the pros and cons either but to me slice seems hacky for some reason. I just know the spread operator is used more commonly but seems there is more ways than one to do the same. All good. I suppose it's personal preference.